src/Sarif/RuleUtilities.cs

```diff
@@ -30,10 +29,10 @@ public static Result BuildResult(FailureLevel level, IAnalysisContext context, R
                 ? ResultKind.Fail
                 : ResultKind.None;
 
-            return BuildResult(level, kind, context, region, ruleMessageId, arguments);
+            return BuildResult(level, kind, context, region, ruleMessageId, reportingDescriptor: null, arguments);
         }
 
-        public static Result BuildResult(FailureLevel level, ResultKind kind, IAnalysisContext context, Region region, string ruleMessageId, params string[] arguments)
+        public static Result BuildResult(FailureLevel level, ResultKind kind, IAnalysisContext context, Region region, string ruleMessageId, ReportingDescriptor reportingDescriptor = null, params string[] arguments)
```

Comment:
>ReportingDescriptor [](start = 141, length = 19)

I'm not sure this is right. The context object already has a reporting descriptor, why should we provide another one? I think it's probably better to create copies of context objects than to depend on the context except for this additional reporting descriptor object (which in your example below can override the context instance).

Reply:
reverted. thanks for reviewing!

---
In reply to: [550509927](https://github.com/microsoft/sarif-sdk/pull/2207#discussion_r550509927) [](ancestors = 550509927)